napari/layers/shapes/shapes.py

```diff
@@ -1997,12 +2045,21 @@ def add(
                 z_index=z_index,
                 n_new_shapes=n_new_shapes,
             )
-            self.events.data(
-                value=self.data,
-                action=ActionType.ADD.value,
-                data_indices=(-1,),
-                vertex_indices=((),),
-            )
+            # This should only emit when programmatically adding as with drawing this leads to premature emit.
```

Comment:
But mode still could be set when someone interacts via API.

Reply:
I adjusted it now by adding the argument `gui` which only is `True` when adding a shape through `_shapes_mouse_bindings`.